### PR TITLE
i#1349 build drltrace parallel to drstrace: clean copy

### PR DESCRIPTION
**Pull request by mxmssh**
drltrace has been copied from dynamorio/clients/drltrace dir to
make further diffs easier and keep our working history clean.

NOTIFY's \n is replaced by NL macros to make runsuite.cmake happy.

### CODE REVIEW

**Comment by derekbruening** [comment]
FYI I'm going to update vs master using github's button here to get the appveyor config for a real windows run and to avoid double-travis -- so you'll need to rebase vs the remote branch to add new commits.

File: drltrace/drltrace.c [hunk 1]
```diff
@@ -0,0 +1,369 @@
...
+            /* Invoke the export to get the real entry: */
+            app_pc (*indir)(void) = (app_pc (*)(void)) cast_to_func(sym->addr);
+            void *drcontext = dr_get_current_drcontext();
+            DR_TRY_EXCEPT(drcontext, {
+                func = (*indir)();
+            }, { /* EXCEPT */
+                func = NULL;
+            });
+            NOTIFY(1, "export %s indirected from "PFX" to "PFX""NL,
+                   sym->name, sym->addr, func);/
```
Comment (derekbruening) [inline drltrace/drltrace.c]: Trailing "/"?

File: drltrace/drltrace.c [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+        else if (sym->is_indirect_code) {
+            /* Invoke the export to get the real entry: */
+            app_pc (*indir)(void) = (app_pc (*)(void)) cast_to_func(sym->addr);
+            void *drcontext = dr_get_current_drcontext();
+            DR_TRY_EXCEPT(drcontext, {
+                func = (*indir)();
+            }, { /* EXCEPT */
+                func = NULL;
+            });
+            NOTIFY(1, "export %s indirected from "PFX" to "PFX""NL,
```
Comment (derekbruening) [inline drltrace/drltrace.c]: I'm curious: why replace \n in the clean copy when nothing else changed and it's not being built?

Empty "" can be elided.

File: drltrace/drltrace.c (hunk 1 above)
Comment (mxmssh) [inline drltrace/drltrace.c]: Done

File: drltrace/drltrace.c (hunk 2 above)
Comment (mxmssh) [inline drltrace/drltrace.c]: It is not being possible to start testing if we have \n. The test script raises an error says that I have to change \n to NL macros for NOTIFY.

**Comment by mxmssh** [comment]
How do I push in this working stream ? I did rebase vs remote branch and then created another commit that you can see in another stream.

File: drltrace/drltrace.c (hunk 2 above)
Comment (derekbruening) [inline drltrace/drltrace.c]: OK, you mean runsuite.cmake's checks, makes sense (maybe say that in the commit msg).

**Comment by derekbruening** [comment]
> How do I push in this working stream ? I did rebase vs remote branch and then created another commit that you can see in another stream.

What are the precise commands that you ran?  Did you run what's in the wiki?  Your original commit has disappeared and I'd like to understand what happened to prevent it in the future.

**Comment by derekbruening** [comment]
> > How do I push in this working stream ? I did rebase vs remote branch and then created another commit that you can see in another stream.
> What are the precise commands that you ran? Did you run what's in the wiki? Your original commit has  disappeared and I'd like to understand what happened to prevent it in the future.

Never mind, the view is confusing, it looks like what we want.  Yes, a separate commit for a new "patchset" of changes for the review.

File: license.txt [hunk 3]
```diff
@@ -484,9 +484,9 @@ DAMAGES.
 ===========================================================================
 BSD License
 
-The drfuzz module is separately licensed under the following BSD license:
+The drfuzz and drltrace modules are separately licensed under the following BSD license:
 
-Copyright (c) 2015 Google, Inc. licensed under the terms of the BSD.  All other rights reserved.
+Copyright (c) 2017 Google, Inc. licensed under the terms of the BSD.  All other rights reserved.
```
Comment (derekbruening) [inline license.txt]: We prefer preserving the range, so 2015-2017

File: drltrace/drltrace.c [hunk 4]
```diff
@@ -197,7 +197,7 @@ iterate_exports(const module_data_t *info, bool add)
                 IF_DEBUG(bool ok =)
                     drwrap_wrap_ex(func, lib_entry, NULL, (void *) sym->name, 0);
                 ASSERT(ok, "wrap request failed");
-                NOTIFY(2, "wrapping export %s!%s @"PFX""NL,
+                NOTIFY(2, "wrapping export %s!%s @"PFX"NL,
```
Comment (derekbruening) [inline drltrace/drltrace.c]: This looks like a syntax error: " is unmatched.  Both "" needed to be removed.  Here and above.

Comment (mxmssh) [inline drltrace/drltrace.c]: I don't understand, how do I combine PFX and NL macros together ?

Comment (derekbruening) [inline drltrace/drltrace.c]: With a space.

**Comment by derekbruening** [comment]
Grrr, Appveyor tests are not in good shape: more failures.  Will override.  Travis Mac is waiting again -- they don't have enough Mac bots.

**Pull request merged**